python/ray/air/checkpoint.py

```diff
@@ -381,6 +444,15 @@ def from_checkpoint(cls, other: "Checkpoint") -> "Checkpoint":
             obj_ref=other._obj_ref,
         )
 
+    @classmethod
+    def _check_type_compatability(cls, serialized_type: Type["Checkpoint"]):
+        if not issubclass(serialized_type, cls):
+            raise ValueError(
+                "The checkpoint data you passed in was created by a "
+                f"`{serialized_type.__name__}` object, but "
+                f"`{serialized_type.__name__}` isn't compatible with `{cls.__name__}`."
+            )
```

Comment:
does the ray version/ray commit come into play here? We should remove it from the metadata if it's not going to be used.

Reply:
we can always add it back later.